File: NProcGen/tests/PonderingProgrammer.Map2dTests/ManhattanFixedSquareMap2dTest.cs
using PonderingProgrammer.Map2d;
using System.Linq;
using Xunit;

namespace PonderingProgrammer.Map2dTests
{
    public class ManhattanFixedSquareMap2dTest
    {
        [Fact]
        public void FindAdjacentCells()
        {
            var map = new ManhattanFixedSquareMap2d<bool>(9, 9);

            Assert.Equal(4, map.FindAdjacentCells(1, 1).Count());
            Assert.Equal(2, map.FindAdjacentCells(0, 0).Count());
            Assert.Equal(3, map.FindAdjacentCells(1, 0).Count());
        }

        [Fact]
        public void ToGraph()
        {
            var map = new ManhattanFixedSquareMap2d<bool>(2, 2);
            var graph = map.ToCellGraph();

            Assert.Equal(4, graph.Nodes.Count);
            Assert.Equal(4, graph.Edges.Count);
        }
    }
}

File: projects/java/map2d/src/test/java/net/ponderingprogrammer/map2d/ManhattanFixedSquareMap2dTest.java
package net.ponderingprogrammer.map2d;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class ManhattanFixedSquareMap2dTest {

    @Test
    void findAdjacentCells() {
        var map = new ManhattanFixedSquareMap2d<Integer>(9, 9);

        Assertions.assertEquals(4, map.findAdjacentCells(1, 1).size());
        Assertions.assertEquals(2, map.findAdjacentCells(0, 0).size());
        Assertions.assertEquals(3, map.findAdjacentCells(1, 0).size());
    }

    @Test
    void toGraph() {
        var map = new ManhattanFixedSquareMap2d<Integer>(2, 2);
        var graph = map.toCellGraph();

        Assertions.assertEquals(4, graph.getNodes().size());
        Assertions.assertEquals(4, graph.getEdges().size());
    }
}